homeassistant/components/websocket_api/commands.py

```diff
@@ -202,3 +204,42 @@ def handle_ping(hass, connection, msg):
     Async friendly.
     """
     connection.send_message(pong_message(msg["id"]))
+
+
+@callback
+@decorators.websocket_command(
+    {
+        vol.Required("type"): "render_template",
+        vol.Required("template"): cv.template,
+        vol.Optional("entity_ids"): cv.entity_ids,
+        vol.Optional("variables"): dict,
+    }
+)
+def handle_render_template(hass, connection, msg):
+    """Handle render_template command.
+
+    Async friendly.
+    """
+    template = msg["template"]
+    template.hass = hass
+
+    variables = msg.get("variables")
+
+    entity_ids = msg.get("entity_ids")
+    if entity_ids is None:
+        entity_ids = list(set(template.extract_entities(variables)))
```

Comment:
`extract_entities` can return `MATCH_ALL`. If that happens, we should return an error as we should not re-render an entity on each state change.

There is no need to wrap the result in `list(set(`  either, that is already done inside extract_entities.

Reply:
Should have tested that first...
Turns out a template like `{{ 42 }}` returns `MATCH_ALL`...

I think the better solution would be to just render such a template once, and not register any entity trackers. This would break tempaltes using `distance()` and `closest()`, but then you could specify an entity manually.

Reply:
Good point. Hmmm ok.

It will give a poor user experience because as a user you don't know why it isn't updating.

Reply:
It would be the same with template sensors and stuff too, though - Right?

Reply:
The problem is where in the stack it runs. A template sensor is outputting values for users, here it is more important that we don't mess up . I think that we should throw an error and instead have the frontend inject an empty list for entity IDs to track.

Reply:
Note that none of this will be relevant once #23590 lands.

Reply:
So let's just keep it as is and not error out .